rules/internal/utils/find.go

```diff
@@ -126,3 +126,26 @@ func GetRepeatedMessageFields(m *desc.MessageDescriptor) []*desc.FieldDescriptor
 
 	return fields
 }
+
+// FindFieldDotNotation attempts to find the field within the given message
+// identified by the dot-notation path e.g. book.name. This is especially useful
+// for resolving path variables in google.api.http and nested
+// fields in google.api.method_signature annotations.
+func FindFieldDotNotation(msg *desc.MessageDescriptor, f string) *desc.FieldDescriptor {
```

Comment:
In essence this works like jquery, right?

So the method description might be:

Returns a field descriptor from a passed in message that corresponds to dot separated path. Example: ...

If path not found the method returns nil.